examples/SMAC4HPO_hyperband_mlp_instances.py

```diff
@@ -0,0 +1,156 @@
+
+        # returns the cross validation accuracy
+        score = cross_val_score(mlp, digits.data, digits.target, cv=int(instance))
+
+    return 1 - np.mean(score)  # Because minimize!
+
+
+logger = logging.getLogger("MLP-example")
+logging.basicConfig(level=logging.INFO)
+logger.info("Running MLP example for SMAC. If you experience "
+            "difficulties, try to decrease the memory-limit.")
+
+# Build Configuration Space which defines all parameters and their ranges.
+# To illustrate different parameter types,
+# we use continuous, integer and categorical parameters.
+cs = ConfigurationSpace()
+
+# We can add multiple hyperparameters at once:
+n_layer = UniformIntegerHyperparameter("n_layer", 1, 5, default_value=2)
+n_neurons = UniformIntegerHyperparameter("n_neurons", 8, 1024, log=True, default_value=10)
+activation = CategoricalHyperparameter("activation", ['logistic', 'tanh', 'relu'],
+                                       default_value='relu')
+solver = CategoricalHyperparameter('solver', ['lbfgs', 'sgd', 'adam'], default_value='adam')
+batch_size = UniformIntegerHyperparameter('batch_size', 30, 300, default_value=200)
+learning_rate = CategoricalHyperparameter('learning_rate', ['constant', 'invscaling', 'adaptive'],
+                                          default_value='constant')
+learning_rate_init = UniformFloatHyperparameter('learning_rate_init', 0.0001, 1.0, default_value=0.001, log=True)
+cs.add_hyperparameters([n_layer, n_neurons, activation, solver, batch_size, learning_rate, learning_rate_init])
+
+# Adding conditions to restrict the hyperparameter space
+# Since learning rate is used when solver is 'sgd'
+use_lr = CS.conditions.EqualsCondition(child=learning_rate, parent=solver, value='sgd')
+# Since learning rate initialization will only be accounted for when using 'sgd' or 'adam'
+use_lr_init = CS.conditions.InCondition(child=learning_rate_init, parent=solver, values=['sgd', 'adam'])
+# Since batch size will not be considered when optimizer is 'lbfgs'
+use_batch_size = CS.conditions.InCondition(child=batch_size, parent=solver, values=['sgd', 'adam'])
+# We can also add  multiple conditions on hyperparameters at once:
+cs.add_conditions([use_lr, use_batch_size, use_lr_init])
+
+# Defining instances (number of folds for CV)
+instances = [['2'], ['3'], ['5'], ['10']]
+
+# SMAC scenario object
+scenario = Scenario({"run_obj": "quality",      # we optimize quality (alternative runtime)
+                     "wallclock-limit": 100,    # max duration to run the optimization (in seconds)
+                     "cs": cs,                  # configuration space
+                     "deterministic": "true",
+                     "limit_resources": False,  # Disables pynisher to pass cutoff directly to the target algorithm.
+                                                # Timeouts have to be taken care within the TA
+                     "cutoff": 10,              # Cutoff denotes the number of epochs for MLP.
```

Comment:
Hm, I find this unintuitive. Shouldn't this be a hyperparameter or a constant, and `cutoff` be a time limit/memory limit?

Reply:
Right now, since `cutoff` is not strictly coupled to pynisher runtime limits, I thought this will be an example of using an arbitrary cutoff (no. of epochs in this case). That is also why this example doesn't use pynisher. But I agree, this does make it a bit unintuitive.